src/Main.js

```diff
@@ -0,0 +1,323 @@
+    this.setState({
+      currentTrackURL: returnBlockURL(item),
+      indexOfCurrentTrack,
+      currentTrackInfo: item,
+      trackIsFromCurrentPlaylist: true,
+      currentTrackPlaylist: this.state.currentOpenPlaylist
+    })
+    this.play()
+  }
+
+
+  // determines if the currently playing/paused track is from the currently
+  // displayed playlist
+  isTrackIsFromCurrentPlaylist = (pl1, pl2) => {
+    if (pl1 && pl2) {
+      return pl1.id === pl2.id ? true : false
+    } else {
+      return true
+    }
+  }
+
+
+  play = () => {
+    this.setState({ isPlaying: true, })
+  }
+
+  pause = () => {
+    this.setState({ isPlaying: false, playerStatus: playerStatus.idle })
+  }
+
+  // if we select a playlist, get it's contents.
+  // then, set it as the current open playlist
+  returnSelectedPlaylist = (playlistSlug) => {
+    this.setState({isCurrentPlaylistLoaded: false})
+    this.API.getFullChannel(playlistSlug)
+      .then(playlist => {
+        const { currentTrackPlaylist } = this.state
+        this.setState({
+          currentOpenPlaylist: playlist,
+          isCurrentPlaylistLoaded: true,
+          trackIsFromCurrentPlaylist: this.isTrackIsFromCurrentPlaylist(currentTrackPlaylist, playlist)
+        })
+      })
+  }
+
+  // update +1 track and index
+  goToNextTrack = () => {
+    const { indexOfCurrentTrack, currentTrackPlaylist } = this.state
+    if (indexOfCurrentTrack + 1 < currentTrackPlaylist.length) {
+      const nextIndex = indexOfCurrentTrack + 1
+      const nextTrack = currentTrackPlaylist.contents[nextIndex]
+      this.handleSongSelection(nextTrack, nextIndex)
+    }
+  }
+
+  //  update -1 track and index
+  goToPreviousTrack = () => {
+    const { indexOfCurrentTrack, currentTrackPlaylist } = this.state
+    if (indexOfCurrentTrack > 0) {
+      const previousIndex = indexOfCurrentTrack - 1
+      const previousTrack = currentTrackPlaylist.contents[previousIndex]
+      this.handleSongSelection(previousTrack, previousIndex)
+    } else if (indexOfCurrentTrack === 0) {
+      this.playerRef.seekTo(0)
+    }
+  }
+
+  returnFullRoute = (currentRoute) => {
+    this.setState({currentRoute})
+  }
+
+  handleOnReady = (e) => {
+    // console.log(e, 'ready')
+  }
+
+  handleOnStart = (e) => {
+    // console.log(e, 'start')
+  }
+
+  handleOnPlay = (e) => {
+    this.setState({playerStatus: playerStatus.playing })
+  }
+
+  handleOnProgress = (e) => {
+    this.setState({ trackProgress: e.playedSeconds })
+  }
+
+  handleOnDuration = (e) => {
+    this.setState({ trackDuration: e })
+  }
+
+  handleOnBuffer = (e) => {
+    this.setState({playerStatus: playerStatus.buffering })
+  }
+
+  handleOnError = (e) => {
+    this.setState({playerStatus: playerStatus.errored })
+    this.goToNextTrack()
+  }
+
+  returnRef = (ref) => {
+    this.playerRef = ref
+  }
+
+  render () {
+    return (
+      <Router>
+        <div className={'w-100 min-vh-100 pa3 pa5-ns'}>
```

Comment:
Btw, I'm not married at all to tachyons. This was just a chance to try it out.

Reply:
I might go back and simplify the styles, I like tachyons too but for certain things ya just gotta write your own classes.